app/models/post.rb

```diff
@@ -1,14 +1,40 @@
 class Post < ActiveRecord::Base
   has_many :comments, dependent: :destroy
+  has_many :votes, dependent: :destroy
   belongs_to :user
   belongs_to :topic
   mount_uploader :image, ImageUploader
+  after_create :create_vote
 
+  def up_votes
+    votes.where(value: 1).count
+  end
 
-  default_scope { order('created_at DESC') }
+  def down_votes
+    votes.where(value: -1).count
+  end
+
+  def points
+    votes.sum(:value)
+  end
+
+  def update_rank
+    age_in_days = (created_at - Time.new(1970,1,1)) / (60 * 60 * 24)
+    new_rank = points + age_in_days
+
+    update_attribute(:rank, new_rank)
+  end
+
+  default_scope { order('rank DESC') }
 
   validates :title, length: { minimum: 5 }, presence: true
   validates :body, length: { minimum: 20 }, presence: true
   validates :topic, presence: true
   validates :user, presence: true
+
+  private
+
+  def create_vote
+    votes.create(value: 1)
```

Comment:
Currently this won't associate a vote directly with the user who created the post, try this (and possibly that'll bring the `post_spec.rb` failures more inline with what the checkpoint reports):

```ruby
user.votes.create(value: 1, post: self)
```